src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.cpp

```diff
@@ -555,44 +929,109 @@ real64 AcousticWaveEquationSEM::explicitStep( real64 const & time_n,
 
     addSourceToRightHandSide( cycleNumber, rhs );
 
+    /// apply the main function to update some of the PML auxiliary variables
+    /// Compute (divV) and (B.pressureGrad - C.auxUGrad) vectors for the PML region
+    if( flagPML>0 )
+    {
+      applyPML( time_n, domain );
+    }
+
+
     /// calculate your time integrators
     real64 const dt2 = dt*dt;
 
     GEOSX_MARK_SCOPE ( updateP );
     forAll< EXEC_POLICY >( nodeManager.size(), [=] GEOSX_HOST_DEVICE ( localIndex const a )
     {
+
       if( freeSurfaceNodeIndicator[a] != 1 )
       {
-        p_np1[a] = p_n[a];
-        p_np1[a] *= 2.0*mass[a];
-        p_np1[a] -= (mass[a]-0.5*dt*damping[a])*p_nm1[a];
-        p_np1[a] += dt2*(rhs[a]-stiffnessVector[a]);
-        p_np1[a] /= mass[a]+0.5*dt*damping[a];
+        if( flagPML==0 )
```

Comment:
A possible improvement here would be to move the `forAll` launch to `AcousticWaveEquationSEMKernels.hpp` and have two versions of it in the `struct`: `launchNonPMLPressureUpdate` with just: 
```
p_np1[a] = ( 2.0*mass[a]*p_n[a]
                       - (mass[a] - 0.5*dt*damping[a])*p_nm1[a]
                       + dt2*(rhs[a] - stiffnessVector[a])
                       ) / (mass[a] + 0.5*dt*damping[a]);
```
and `launchPMLPressureUpdate` (feel free to come up with better names) with what comes after the `else`. 
The main benefit would be to have the `if( flagPML )` outside the kernel, which would be cleaner.

Reply:
I have split the `forAll` launch and moved the `if (usePML)` outside the launch. But II kept the code in the function `AcousticWaveEquationSEM::explicitStep`.